handler/src/main/java/io/netty/handler/stream/ChunkedWriteHandler.java

```diff
@@ -285,6 +285,7 @@ public void operationComplete(ChannelFuture future) throws Exception {
                             } else {
                                 currentWrite.progress(chunks.progress(), chunks.length());
                                 currentWrite.success(chunks.length());
+                                closeInput(chunks);
```

Comment:
@lutovich Can we move this call out of the `if...else` block?

Reply:
@lutovich seems like this was missed ?

Reply:
I changed it back on purpose in the second commit to make sure listener only invokes `ChunkedInput#close()` on failure. On success it also needs to read progress and length from the input. Does this make sense?

Reply:
ah ok yeah.